### PR TITLE
fix(medusa): Include adjustments when authorizing payment

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
Include relation `adjustments` on cart in `CartService.authorizePayment`

**Why**
Adjustments were not taken into account when calculating the total amount of a payment prior to creating it

**Tests**
Integration test that asserts correct total, subtotal, and discount total as well as payment object upon cart completion

**Thoughts**
We should think of a more scalable way of making sure that we always include relations required to calculate totals, such that we don't end up in a similar situation again.

Though, I've decided to prioritize pushing a quick fix now over the scalable solution given the high severity level of the issue.

### CODE REVIEW

**Review by pKorsholm — APPROVED** [review]
LGTM! definitely support that we should just get something working up and then update the solution at a later point 👍

File: integration-tests/api/helpers/cart-seeder.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ module.exports = async (connection, data = {}) => {
   const r = manager.create(Region, {
     id: "test-region",
     name: "Test Region",
+    payment_providers: [{ id: "test-pay" }],
```
Comment (adrien2p) [inline integration-tests/api/helpers/cart-seeder.js:53]: Very nice 👌 did it raised a lot?

File: packages/medusa/src/services/line-item-adjustment.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ class LineItemAdjustmentService extends BaseService {
       const lineItemAdjustmentRepo: LineItemAdjustmentRepository =
         manager.getCustomRepository(this.lineItemAdjustmentRepo_)
 
-      const lineItemAdjustment = await lineItemAdjustmentRepo.create(data)
+      const lineItemAdjustment = lineItemAdjustmentRepo.create(data)
```
Comment (adrien2p) [inline packages/medusa/src/services/line-item-adjustment.ts:99]: Nicely spotted 🤞

**Review by adrien2p — APPROVED** [review]
LGTM 🚀 we should indeed merge that one. Maybe later, we should have a const that represents the relations needed to compute a total and use that const anywhere we need it. That way, if we include new relations as part of the refacto it will add the relations everywhere it is needed?

in Sequelize, we had something called a scope, it was something link to the model and when you were qurying you could include a scope that was a partial object containing common constraints and relations part of the same scope

File: integration-tests/api/helpers/cart-seeder.js (hunk 1 above)
Comment (olivermrbl) [reply integration-tests/api/helpers/cart-seeder.js:53]: A couple of tests were redundant without this addition, but should be good now 👍

**Review by srindom — APPROVED** [review]
LGTM

**Comment by olivermrbl** [comment]
Changing base to `master` to allow for a quick fix release without the newly added BatchJob API (good thing we are soon to scope out feature flags 😅 )

cc @pKorsholm @adrien2p @srindom

**Review by zakariaelas — APPROVED** [review]

**Review by adrien2p — APPROVED** [review]

**Comment by olivermrbl** [comment]
Git caused me a bit of issues, when I tried to change base + history 😅 

Decided to instead create a new PR [here](https://github.com/medusajs/medusa/pull/1697).

**Pull request closed without merging**